packages/checkout/core/src/providers/base-provider/base-provider.types.ts

```diff
@@ -8,8 +8,8 @@ export interface Customer {
   phoneNumber: string
   document: {
```

Comment:
I believe here it should be added as optional because it's the parameter configuration interface in the checkout

```suggestion
  document?: {
```